packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -18,6 +19,29 @@ moduleFor('Components test: curly components', class extends RenderingTest {
     this.assertComponentElement(this.firstChild, { content: 'hello' });
   }
 
+  // Note this functionality seems weird
```

Comment:
Makes sense. Added this test. The only thing is that an `EmberError` gets thrown so `expectAssertion` is not the correct assertion as it won't be stripped in prod builds.